sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/QueryIdGenerator.kt

```diff
@@ -0,0 +1,18 @@
+package com.squareup.sqldelight.core.compiler
+
+/**
+ * When generating IDs for queries, we need to ensure they are done in a deterministic way
+ * The ordering of database generation is non-deterministic so we scope the ids to a database
+ * Secondly, we need to ensure the counter is reset between gradle invocations.
+ * This class should be instantiated at the beginning of code generation
+ *
+ * Note: This implementation is not thread-safe and should therefore not be shared between gradle tasks
```

Comment:
can we use an `AtomicInteger` here and make it thread safe? one of my next goals is letting this all be concurrent

Reply:
yea, ours right now is broken so it definitely needs attention.

Reply:
Using an atomic integer is going to make generation non-deterministic again if concurrency is introduced

Reply:
uhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh tru

Reply:
Correct. My assumption is that improving concurrency would be on the module level instead of task level

Reply:
right - concurrency between modules is what i want to fix, but task concurrency at some point would be good too